// Copyright 2010 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Author: jmarantz@google.com (Joshua Marantz)
// Protobufs to help with http caching

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package net_instaweb;

message NameValue {
  required string name = 1;
  required string value = 2;
};

message HttpResponseHeaders {
  optional int32 status_code = 1;
  optional string reason_phrase = 2;
  optional int32 minor_version = 3;
  optional int32 major_version = 4;
  optional int64 expiration_time_ms = 5;
  optional int64 timestamp_ms = 6;
  optional bool cacheable = 7;
  optional bool proxy_cacheable = 8;
  repeated NameValue header = 9;
};
